juju/unit.py

```diff
@@ -108,21 +109,58 @@ def resolved(self, retry=False):
         :param str command: The command to run
         :param int timeout: Time to wait before command is considered failed
 
+        Returns a tuple containing the stdout, stderr, and return code
+        from the command.
+
         """
         action = client.ActionFacade()
         action.connect(self.connection)
 
         log.debug(
             'Running `%s` on %s', command, self.name)
 
-        # TODO this should return an Action
-        return await action.Run(
+        action_status = asyncio.Queue(loop=self.model.loop)
```

Comment:
While this approach works, we have a helper in place to make it more concise. Something like this;

```python
res = await action.Run(
    [],
    command,
    [],
    timeout,
    [self.name],
)
tag = res.results[0].action.tag

def predicate(delta):
    if delta.data['status'] in ('completed', 'error'):
        return True

return await self.model._wait_for_new('action', tag, predicate)
```

There are some other examples of _wait_for_new() in the codebase.